homeassistant/components/pooldose/sensor.py

```diff
@@ -0,0 +1,228 @@
+    ),
+    PooldoseSensorEntityDescription(
+        key="orp_calibration_slope",
+        translation_key="orp_calibration_slope",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.VOLTAGE,
+        suggested_display_precision=2,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=UnitOfElectricPotential.MILLIVOLT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PooldoseConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up PoolDose sensor entities from a config entry."""
+    # Assert for type checker that unique_id is not None
+    if TYPE_CHECKING:
+        assert config_entry.unique_id is not None
+
+    coordinator = config_entry.runtime_data.coordinator
+    client = config_entry.runtime_data.client
+    serial_number = config_entry.unique_id
+    device_properties = config_entry.runtime_data.device_properties
+
+    available = client.available_sensors()
+
+    async_add_entities(
+        [
+            PooldoseSensor(
+                coordinator,
+                description,
+                serial_number,
+                device_properties,
+            )
+            for description in SENSOR_DESCRIPTIONS
+            if description.key in available
+        ]
+    )
+
+
+class PooldoseSensor(PooldoseEntity, SensorEntity):
+    """Sensor entity for the Seko PoolDose Python API."""
+
+    entity_description: PooldoseSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PooldoseCoordinator,
+        description: PooldoseSensorEntityDescription,
+        serial_number: str,
+        device_properties: dict[str, Any],
+    ) -> None:
+        """Initialize a PoolDose sensor entity."""
+        super().__init__(
+            coordinator,
+            serial_number,
+            device_info(device_properties, serial_number),
+            entity_description=description,
+        )
+
+    @property
+    def native_value(self) -> float | int | str | None:
+        """Return the current value of the sensor."""
+        sensor_data = self.coordinator.data.get(self.entity_description.key)
+        if isinstance(sensor_data, (list, tuple)) and sensor_data:
+            return sensor_data[0]
+        return None
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return the unit of measurement."""
+        # Use static unit if defined
+        if self.entity_description.native_unit_of_measurement:
+            return self.entity_description.native_unit_of_measurement
+
+        # For temperature, get unit from API data
+        if self.entity_description.key == "temperature" and self.coordinator.data:
+            sensor_data = self.coordinator.data.get("temperature")
+            if isinstance(sensor_data, (list, tuple)) and len(sensor_data) > 1:
+                return sensor_data[1]  # °C or °F
+
+        return None
```

Comment:
```suggestion
    @property
    def native_unit_of_measurement(self) -> str | None:
        """Return the unit of measurement."""
        if self.entity_description.key == "temperature" and self.coordinator.data:
            sensor_data = self.coordinator.data.get("temperature")
            if isinstance(sensor_data, (list, tuple)) and len(sensor_data) > 1:
                return sensor_data[1]  # °C or °F

        return super().native_unit_of_measurement
```
I would maybe recommend moving the temperature uom to a function in the entity description, but since this only happens once I don't think this is a big deal